pallets/loans/src/entities/loans.rs

```diff
@@ -243,26 +243,21 @@ impl<T: Config> ActiveLoan<T> {
 		}
 	}
 
-	pub fn principal(&self, pool_id: T::PoolId) -> Result<T::Balance, DispatchError> {
-		match &self.pricing {
-			ActivePricing::Internal(inner) => Ok(self
-				.total_borrowed
-				.ensure_sub(self.total_repaid.principal)?),
-			ActivePricing::External(inner) => {
-				let maturity = self.maturity_date();
-				inner.outstanding_principal(pool_id, maturity)
-			}
-		}
+	pub fn principal(&self) -> Result<T::Balance, DispatchError> {
+		Ok(self
+			.total_borrowed
+			.ensure_sub(self.total_repaid.principal)?)
```

Comment:
Why the change? Does this mean, that the gains we are making through price increases are repayed as unscheduled?

Reply:
It seems like you have reviewed something outdated

Reply:
But I get your point, it should be just the borrowed amount, not taking into account the repaid one, right?

Reply:
Now its compeltly gone, right? But looks good now

Reply:
It's still there, I need to remove the `.ensure_sub(self.total_repaid.principal)?` part, to not reduce the cashflow expected amount with the paid amount